src/main/python/pgshovel/replication/streams/kafka.py

```diff
@@ -40,12 +54,44 @@ def consume(self, state):
             consumer.seek(delta, 1)
             assert consumer.offsets[0] == offset
 
-        for offset, message in consumer:
-            decoded = self.codec.decode(message.value)
-            state = validate_state(state, offset, decoded)
+        for offset, message in self.__primed(consumer, state):
```

Comment:
> that the validation would still fail as the sequence number did not increase by 1, but increased by the number of messages that were skipped

The transaction state validation should be independent of the message sequence validation, and I don't think it's a good idea to rely on one to ensure valid behavior of the other.

In this case, if the `Begin` was sent by the relay but not persisted by Kafka, this would be caught, but if the `Begin` was not sent by the relay (due to an unforeseen implementation error) this would not be caught.